### ISSUE TITLE
Unresponsive

### ISSUE THREAD

**Issue by cdglasz** [kmackenzieii/ai-midterm#17]
The agent will occasionally become unresponsive. This isn't affected by killing the agent in-game or closing the game window, so it may be an infinite loop; Kyle suspects it's related to using too many rays
Cause needs to be located

**Comment by kmackenzieii** [comment]
There's an infinite loop somewhere in your indiscretize function. If you
wait long enough without killing the quagent it overflows the heap and
causes a crash.

**Comment by cdglasz** [comment]
Thanks, Kyle, I'll sort that out now

**Comment by cdglasz** [comment]
It seems that shooting it almost always returns it to regular operation. Either of you know why that could be?

**Comment by cdglasz** [comment]
I can't find a place where there would be an infinite loop in the indiscritize method. The only situation I can think of would be if there was a wall between the goal node and every other node in the path, but that doesn't make sense, A* couldn't produce such a path

**Comment by kmackenzieii** [comment]
So everything looks great. I actually ran it with the parasite and it was able to explore the entire space. I didn't see it become unresponsive until the entire room had been explored. In all I don't think its really an issue.

**Comment by cdglasz** [comment]
Really? When you get the chance, test it a few times, just to make sure that it doesn't happen. We should also figure out which quagent (parasite, berserker, etc) performs the best. If I have time tomorrow I'll do a few trials and time how long it takes them to map the room.
Does the whole picking up tofu thing work? I haven't actually tested it at all

**Comment by kmackenzieii** [comment]
Yup. Scattered 6 pieces of tofu and it finds them all. Since the parasite
is capable of navigating I say we use it since it's so fast. Second I would
say is soldier. The others are slower and I think have a bigger hit box. We
need to write up a report for when we submit it too. I suggest a Google doc
so we can all edit at once.

**Comment by cdglasz** [comment]
I agree with that. I'm still going to do some time trials though, just in case one performs better for some reason, but yeah, I think parasite will probably end up being our best bet. Also yeah, that would work.

**Comment by kmackenzieii** [comment]
I just did it again and it got all the tofu and explored the whole room in
about 2:30. Don't think it can get much better. Again, it didn't hang until
the whole room was explored anyway.
